flytepropeller/pkg/controller/workflow/executor.go

```diff
@@ -282,13 +287,16 @@ func (c *workflowExecutor) TransitionToPhase(ctx context.Context, execID *core.W
 			wfEvent.Phase = core.WorkflowExecution_RUNNING
 			wStatus.UpdatePhase(v1alpha1.WorkflowPhaseRunning, "Workflow Started", nil)
 			wfEvent.OccurredAt = utils.GetProtoTime(wStatus.GetStartedAt())
-		case v1alpha1.WorkflowPhaseHandlingFailureNode:
-			fallthrough
 		case v1alpha1.WorkflowPhaseFailing:
 			wfEvent.Phase = core.WorkflowExecution_FAILING
 			wfEvent.OutputResult = convertToExecutionError(toStatus.Err, previousError)
 			wStatus.UpdatePhase(v1alpha1.WorkflowPhaseFailing, "", wfEvent.GetError())
 			wfEvent.OccurredAt = utils.GetProtoTime(nil)
+		case v1alpha1.WorkflowPhaseHandlingFailureNode:
```

Comment:
Can you explain why we need another phase instead of just applying this during the `WorkflowPhaseFailing` handle?

Reply:
since we handle failure node in `WorkflowPhaseHandlingFailureNode` phase here
https://github.com/flyteorg/flyte/blob/ad0597cd2b9854b23c24e77260e045038532281f/flytepropeller/pkg/controller/workflow/executor.go#L422-L431.
we are able to move it to `WorkflowPhaseFailing` as well

Reply:
Ah OK, so you didn't add the `WorkflowPhaseHandlingFailureNode` just implemented it correctly?